src/Oval.sol

```diff
@@ -39,11 +38,31 @@ abstract contract Oval is DiamondRootOval {
      * @notice Returns latest data from source, governed by lockWindow controlling if returned data is stale.
      * @return answer The latest answer in 18 decimals.
      * @return timestamp The timestamp of the answer.
+     * @return roundId The roundId of the answer.
      */
-    function internalLatestData() public view override returns (int256, uint256) {
+    function internalLatestData() public view override returns (int256, uint256, uint256) {
```

Comment:
This change makes the implementation very directly coupled to chainlink; we are no longer not opinionated about the implementation of the source & destination relationship and pull the roundId throughout all adapters, from the source to destination. I understand why we made the change like this: its easier to not have to fetch this directly from the destination adapter. However, this coupling makes implementations of Oval that do not use chainlink more clumsy.

Do you think there is an implementation that leaves the notion of "internal data" as a `[int256, uint256]` for value and time and enables the retrieval of round Id from the destination adapter? due to these all compressing into one contract (inheritance structure we use via diamond) it might be posible to put the responsibility for this retrival within the destination adapter?

I'm curious what you think about this perspective.

Reply:
agree, this couples the implementation more with Chainlink. but I'm not sure its worth reimplementing `internalLatestData` in the destination adapter to pass through round information as this method is also used to control access to price data via unlocks.

Also, we could implement the concept of rounds in any of other source adapters that don't have native history, through snapshotting - then this would not be  coupled with chainlink as all adapters could expose this

Reply:
Agree with this comment. Maybe the internalLatestData third returned value could be a bytes value carrying ancillary data that the destination adapter knows how to consume.